### ISSUE TITLE
[BUG]: BGR input is fed to all models

### ISSUE THREAD

**Issue by PavloFesenko** [serengil/deepface#1457]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [x] I have updated to the latest version of the packages.
- [x] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.93

### Python version

3.11.11

### Operating System

Ubuntu 24.04 (Linux Mint 22)

### Dependencies

tensorflow==2.19.0
tf-keras==2.19.0

### Additional Info

Right before feeding the image to the face embedding model, DeepFace transforms the image from RGB to BGR which is weird because models are typically trained on the RGB input. Also, I'm getting noticeably better results with DeepFace when manually reversing the color order for the input image arrays.

https://github.com/serengil/deepface/blob/96a7b98f337c57d0080203b708dcef3ceccca277/deepface/modules/representation.py#L138-L139

Overall, I find the DeepFace code around BGR and RGB extremely confusing which likely led to this issue. I would be happy to submit a PR that will fix it and would suggest the following approach:

- The input image arrays for all methods `analyze`, `represent`, etc. should be in RGB and not in BGR.
- The only BGR-to-RGB transformation would be required when reading images using OpenCV which was commented out for some reason. https://github.com/serengil/deepface/blob/96a7b98f337c57d0080203b708dcef3ceccca277/deepface/commons/image_utils.py#L126-L127
- All image outputs should be in RGB before they are fed to the models but additional audit should be done to confirm that the model input is RGB because a few might require BGR (for example, for the OpenCV detection model).

**Comment by serengil** [comment]
First things first, you are more than welcome!

But my experiments show that BGR is offering higher accuracy score than RGB.

For instance, current accuracy score of [this test](https://github.com/serengil/deepface/blob/master/tests/test_verify.py#L72) is 99.33333333333333. If you can get a higher score than this, then I will be happy to merge your PR then.

**Comment by serengil** [comment]
I am not interested in nearest neighbors at all. If the accuracy over LFW is higher in RGB, then we can say that is better. If you provide accuracy scores for BGR and RGB, we can discuss this further.

**Comment by serengil** [comment]
Closing this because this is not a bug, an intended behavior.

You can still share your outcomes in this issue and we can continue to discuss.
